### PR TITLE
Kh dev admin edit order status

### LINKED ISSUE #113: Story 18: Admin Orders
As an Admin
      When I visit the dashboard
      Then I can see a listing of all orders
      And I can see the total number of orders for each status **("Ordered", "Paid", "Cancelled", "Completed")**
      And I can see a link for each individual order
      And I can filter orders to display by each status type  **("Ordered", "Paid", "Cancelled", "Completed")**
      And I have links to transition between statuses
        - I can click on "cancel" on individual orders which are "paid" or "ordered"
        - I can click on "mark as paid" on orders that are "ordered"
        - I can click on "mark as completed" on orders that are "paid"

### PR DESCRIPTION
**Pull request by kheppenstall**
-Add dropdown in admin dashboard to edit status of orders
-Feature tests for all order updates in story

Closes #113 and Story 18

@wlffann @akintner @tmikeschu Needs 1 reviewer.

screen shot 2017-01-07 at 9 43 53 pm

### CODE REVIEW

File: spec/features/admin/admin_edits_order_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  context 'visit admin dashboard' do
+    context 'admin' do
+      scenario 'change order status from ordered to cancelled' do
+        create(:order_with_items, status: 0)
+        visit admin_dashboard_path
+
+        click_on 'cancel'
+
+        expect(Order.cancelled.count).to eq 1
```
Comment (tmikeschu) [inline spec/features/admin/admin_edits_order_spec.rb]: Can you throw an expectation of 0 before the cancel? I'm partial explicitly showing what a link does.

File: app/controllers/admin/base_controller.rb [hunk 2]
```diff
@@ -1,12 +1,8 @@
 class Admin::BaseController < ApplicationController
   before_action :require_admin
 
-  def require_admin
-    render file: "/public/404" unless current_admin?
-  end
-
   def dashboard
-    @orders = Order.by_status(params[:status])
+    @orders = Order.by_status(params[:status]).order('updated_at DESC')
```
Comment (tmikeschu) [inline app/controllers/admin/base_controller.rb]: Can the ordering happen at the model level?

File: spec/features/admin/admin_edits_order_spec.rb (hunk 1 above)
Comment (kheppenstall) [inline spec/features/admin/admin_edits_order_spec.rb]: Definitely. Will do.

File: app/controllers/admin/base_controller.rb (hunk 2 above)
Comment (kheppenstall) [inline app/controllers/admin/base_controller.rb]: @tmikeschu Good suggestion.

**Comment by kheppenstall** [comment]
@tmikeschu Comments addressed and ready to merge!

**Comment by tmikeschu** [comment]
@kheppenstall thanks for those little fixes!

**Pull request merged**